common/logisticspipes/utils/gui/InputBar.java

```diff
@@ -43,6 +43,8 @@ public InputBar(FontRenderer fontRenderer, LogisticsBaseGuiScreen screen, int le
 
 	public InputBar(FontRenderer fontRenderer, LogisticsBaseGuiScreen screen, int left, int top, int width, int height, boolean isActive, boolean numberOnly, Align align) {
 		super(0, fontRenderer, left, top, width, height);
+		screen.onGuiEvents.add(this);
+		Keyboard.enableRepeatEvents(true);
```

Comment:
Don't do that in the constructor, since just creating an instance of InputBar will modify the keyboard state. You don't want this.

Put that call in the GUI classes itself, where an input bar is used. That's how it's done for vanilla guis aswell (unless they've changed that).

Also, unless you're already doing that somewhere, you need to call Keyboard.enableRepeatEvents(false); when the gui is closed (GuiScreen::onGuiClosed, I think the method is called)

Reply:
vanilla guis initGui()
    {
        Keyboard.enableRepeatEvents(true);
.........
        this.nameEdit = new GuiTextField(...);
        this.nameEdit.setFocused(true);
        this.nameEdit.setText(s);
    }

Keyboard.enableRepeatEvents(false); is already called in event subscription on closeGui() of parent

Digging a little deeper found there could be another enableRepeatEvents(false) calls between them.
Therefore enableRepeatEvents(true) will move to onKeyboardInput event, as close to keyboard loop as possible ?

Reply:
No, put it in the same location as vanilla guis. So initGui().

Reply:
There seemed to be a better place: reposition(), but not every initGui() calls it.
Should I move to LogisticsBaseGuiScreen.initGui/closeGui() or 
every inheritor where InputBar is used (12), increasing entropy?

Reply:
on LogisticsBaseGuiScreen. Due to possible nesting SubGuiScreen

Reply:
well, since SubGuiScreen is again a GuiScreen, also in initGui of the SubGuiScreens that use it (which I'm not seeing any that do, I think), and not in LogisticsBaseGuiScreen.

How is reposition() a better place for it? It does something completely unrelated...

Reply:
closeGui() of nested SubGuiScreen should not call enableRepeatEvents(false), it is difficult to determine the last.
reposition() was called every time a simple window was shown and returned after overlapping by others, such as JEI.
Now I already doubt and I put calls in vanilla style, a little worried about the size of the diff.

Reply:
Yep, that is true. Just call initGui of the main screen again when the subscreen closes, it doesn't hurt.